src/common/schema-derived-types.ts

```diff
@@ -18,25 +19,28 @@ type RecursivePartial<T> = T extends object
 type JsonSchemaTypeToControlOptions<
   T,
   // K extends keyof T & string,
-> = T extends { type: infer U }
-  ? U extends "object" // ObjectControlOptions goes here
-    ? unknown
-    : U extends "string"
-      ? TextControlOptions
-      : U extends "number" | "integer"
-        ? NumericControlOptions
-        : U extends "array"
-          ? ArrayControlOptions
-          : U extends "boolean"
-            ? unknown // BooleanControlOptions goes here
-            : unknown
-  : T extends { anyOf: unknown }
-    ? AnyOfControlOptions
-    : T extends { oneOf: unknown }
-      ? OneOfControlOptions
-      : unknown
+> = T extends { enum: unknown }
```

Comment:
Had to add this before `type` because `enum` and `type` can coexist at the same level with `type` referring to the type of the values in the enum.